### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by jagrg** [emacs-helm/helm-dictionary#24]
With `helm-dictionary-20160309.910`, I get the following message when calling helm-dictionary: `helm-p-candidates-file-init: Opening input file: no such file or directory, /usr/share/trans/de-en`

It seems like `helm-dictionary-database` is only accepting the default value (which is "/usr/share/trans/de-en"). Try changing `helm-dictionary-database` to a different value and see if helm-dictionary uses that value instead? This may be related to the `helm-plugin`'s package.

**Comment by tmalsburg** [comment]
@thierryvolpiatto is this related to your changes?  I haven't made any other changes in helm-dictionary recently.

**Comment by tmalsburg** [comment]
Oh, wait.  @jagrg did you restart Emacs after updating helm-dictionary from MELPA?  What is the value of your `helm-dictionary-database`?

**Comment by jagrg** [comment]
I did, yes. I also tried with Emacs -Q.

(setq helm-dictionary-database "~/.emacs.d/en-pt-enwiktionary.txt")

>>>>> On 10 Mar 2016 at 17:02, Titus von der Malsburg <notifications@github.com> wrote:
> Oh, wait. @jagrg did you restart Emacs after updating
> helm-dictionary from MELPA? What is the value of your
> `helm-dictionary-database`?

> --- Reply to this email directly or view it on GitHub:
> https://github.com/emacs-helm/helm-dictionary/issues/24#issuecomment-194956995

-- 
Jonathan

**Comment by thierryvolpiatto** [comment]
Titus von der Malsburg <notifications@github.com> writes:

> @thierryvolpiatto is this related to your changes?

No, I don't think so, IMO the error comes that the user did not restart
emacs after compiling (i.e helm-plugin is no more used).

-- 
Thierry

**Comment by tmalsburg** [comment]
@jagrg I agree with @thierryvolpiatto.  If `helm-dictionary-database` was really defined to be `~/.emacs.d/en-pt-enwiktionary.txt`, helm-dictionary simply cannot access `/usr/share/trans/de-en` because that values shouldn't even be stored in memory.  Hence, there must be problem with that variable.  Please try:

    (message helm-dictionary-database)

**Comment by jagrg** [comment]
Ok, fixed. The problem had to do with my `use-package` configuration. 

This works:

```
(use-package helm-dictionary
  :init
  (setq helm-dictionary-database "~/.emacs.d/en-pt-enwiktionary.txt"))
```

But this doesn't:

```
(use-package helm-dictionary
  :config
  (setq helm-dictionary-database "~/.emacs.d/en-pt-enwiktionary.txt"))
```

**Comment by thierryvolpiatto** [comment]
Jonathan Gregory <notifications@github.com> writes:

> I did, yes. I also tried with Emacs -Q.
>
> (setq helm-dictionary-database "~/.emacs.d/en-pt-enwiktionary.txt")

This will not fix your problem, the error according from your initial
message comes from `helm-p-candidates-file-init` which is no more used
by helm-dictionary, that's mean your old code is still loaded, so or you
have not restarted emacs after upgrading or you had a problem after
compilation.
Another answer would be you have a old helm-dictionary.el(c) file laying
somewhere.

I suggest that you recompile helm-dictionary.el from emacs-helm.sh and
then restart your main emacs.

This is typically the problem we had before compiling our files with
async.

Oh, but wait helm-dictionary is not compiled async by default isn't it ?

-- 
Thierry
